### ISSUE TITLE
$] and $OLD_PERL_VERSION deprecated?

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14675]
Migrated from [rt.perl.org#124402](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=124402) (status was 'resolved')

Searchable as RT124402$

**Comment by p5pRT** [comment]
### From the.rob.dixon@gmail.com

This is a bug report for perl from the\.rob\.dixon@&#8203;gmail\.com\,
generated with the help of perlbug 1\.40 running under perl 5\.20\.0\.




---
The old interpreter version variable $\] is described differently in
different versions of perlvar\.

\- Version 20\.1 has it under the list of "Deprecated and removed
variables"\, and doesn't mention $OLD\_PERL\_VERSION\. It adds "Deprecated
in Perl v5\.12\.0\."

\- Version 12\.0 lists it as one of the regular variables\, but
gives no English equivalent\. There is no mention of deprecation

\- Version 14\.0 is the first appearance of the "Deprecated and removed
variables" section\, and $\] is listed there\, together with the English
equivalent $OLD\_PERL\_VERSION\. Here it says "Deprecated in Perl 5\.6\."

I can find no mention of the deprecation in any perldelta\, but $\]
isn't the easiest thing to search for\.

Clearly the accuracy of anything but the most recent version of
perlvar isn't a valid topic for a perlbug\, but it would be nice if the
next version clarified the true situation\.



---


---
Flags&#8203;:
&nbsp;    category=docs
&nbsp;    severity=low


---
Site configuration information for perl 5\.20\.0&#8203;:

Configured by strawberry\-perl at Fri May 30 23&#8203;:27&#8203;:10 2014\.

Summary of my perl5 \(revision 5 version 20 subversion 0\) configuration&#8203;:

&nbsp;  Platform&#8203;:
&nbsp;    osname=MSWin32\, osvers=6\.2\, archname=MSWin32\-x86\-multi\-thread\-64int
&nbsp;    uname='Win32 strawberry\-perl 5\.20\.0\.1 \#1 Fri May 30 23&#8203;:25&#8203;:11 2014 i386'
&nbsp;    config\_args='undef'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=undef
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    use64bitint=define\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, ccflags =' \-s \-O2 \-DWIN32  \-DPERL\_TEXTMODE\_SCRIPTS
\-DPERL\_IMPLICIT\_CONTEXT \-DPERL\_IMPLICIT\_SYS \-DUSE\_PERLIO \-fwrapv
\-fno\-strict\-aliasing \-mms\-bitfields'\,
&nbsp;    optimize='\-s \-O2'\,
&nbsp;    cppflags='\-DWIN32'
&nbsp;    ccversion=''\, gccversion='4\.8\.2'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=12345678
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long long'\, ivsize=8\, nvtype='double'\, nvsize=8\,
Off\_t='long long'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='g\+\+'\, ldflags ='\-s \-L"C&#8203;:\\STRAWB~3\\perl\\lib\\CORE" \-L"C&#8203;:\\STRAWB~3\\c\\lib"'
&nbsp;    libpth=C&#8203;:\\STRAWB~3\\c\\lib C&#8203;:\\STRAWB~3\\c\\i686\-w64\-mingw32\\lib
C&#8203;:\\STRAWB~3\\c\\lib\\gcc\\i686\-w64\-mingw32\\4\.8\.2
&nbsp;    libs=\-lmoldname \-lkernel32 \-luser32 \-lgdi32 \-lwinspool \-lcomdlg32
\-ladvapi32 \-lshell32 \-lole32 \-loleaut32 \-lnetapi32 \-luuid \-lws2\_32
\-lmpr \-lwinmm \-lversion \-lodbc32 \-lodbccp32 \-lcomctl32
&nbsp;    perllibs=\-lmoldname \-lkernel32 \-luser32 \-lgdi32 \-lwinspool
\-lcomdlg32 \-ladvapi32 \-lshell32 \-lole32 \-loleaut32 \-lnetapi32 \-luuid
\-lws2\_32 \-lmpr \-lwinmm \-lversion \-lodbc32 \-lodbccp32 \-lcomctl32
&nbsp;    libc=\, so=dll\, useshrplib=true\, libperl=libperl520\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_win32\.xs\, dlext=xs\.dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags='\-mdll \-s \-L"C&#8203;:\\STRAWB~3\\perl\\lib\\CORE"
\-L"C&#8203;:\\STRAWB~3\\c\\lib"'




---
@&#8203;INC for perl 5\.20\.0&#8203;:
&nbsp;    C&#8203;:/Strawberry/perl/site/lib/MSWin32\-x86\-multi\-thread\-64int
&nbsp;    C&#8203;:/Strawberry/perl/site/lib
&nbsp;    C&#8203;:/Strawberry/perl/vendor/lib
&nbsp;    C&#8203;:/Strawberry/perl/lib
&nbsp;    \.



---
Environment for perl 5\.20\.0&#8203;:
&nbsp;    HOME=C&#8203;:\\Windows\\system32\\config\\systemprofile
&nbsp;    LANG \(unset\)
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=C&#8203;:\\Program Files \(x86\)\\Common Files\\Intel\\Shared
Files\\cpp\\bin\\Intel64;C&#8203;:\\Program Files\\Haskell
Platform\\2014\.2\.0\.0\\lib\\extralibs\\bin;C&#8203;:\\Program Files\\Haskell
Platform\\2014\.2\.0\.0\\bin;C&#8203;:\\ProgramData\\Oracle\\Java\\javapath;c&#8203;:\\Program
Files \(x86\)\\NVIDIA Corporation\\PhysX\\Common;C&#8203;:\\Program Files
\(x86\)\\ImageMagick\-6\.8\.9\-Q16;C&#8203;:\\Program Files \(x86\)\\Haskell
Platform\\2013\.2\.0\.0\\lib\\extralibs\\bin;C&#8203;:\\Program Files \(x86\)\\Haskell
Platform\\2013\.2\.0\.0\\bin;C&#8203;:\\Program Files \(x86\)\\PC Connectivity
Solution\\;C&#8203;:\\Program Files\\Common Files\\Microsoft Shared\\Windows
Live;C&#8203;:\\Program Files
\(x86\)\\PHP\\;C&#8203;:\\Windows\\system32;C&#8203;:\\Windows;C&#8203;:\\Windows\\System32\\Wbem;C&#8203;:\\Windows\\System32\\WindowsPowerShell\\v1\.0\\;C&#8203;:\\Program
Files\\jEdit;C&#8203;:\\usr\\local\\ppt\\bin;C&#8203;:\\Program Files
\(x86\)\\GnuWin32\\bin;C&#8203;:\\Program Files \(x86\)\\Smart
Projects\\IsoBuster;C&#8203;:\\Program Files \(x86\)\\Oracle\\Berkeley DB 11gR2
5\.3\.15\\bin;C&#8203;:\\Program Files \(x86\)\\Git\\cmd;C&#8203;:\\Program Files
\(x86\)\\Bazaar;C&#8203;:\\Program Files \(x86\)\\Lua\\5\.1;C&#8203;:\\Program Files
\(x86\)\\Lua\\5\.1\\clibs;C&#8203;:\\Program
Files\\TortoiseSVN\\bin;C&#8203;:\\MediaInfoCLI;C&#8203;:\\Program Files
\(x86\)\\MKVToolNix;C&#8203;:\\Program Files \(x86\)\\Subversion\\bin;C&#8203;:\\Program
Files \(x86\)\\QuickTime\\QTSystem\\;C&#8203;:\\Program Files\\Microsoft Network
Monitor 3\\;C&#8203;:\\Program Files\\Calibre2\\;C&#8203;:\\Program Files
\(x86\)\\MySQL\\MySQL Utilities 1\.3\.4\\;C&#8203;:\\Program Files \(x86\)\\Common
Files\\Acronis\\SnapAPI\\;C&#8203;:\\Program Files \(x86\)\\Windows Kits\\8\.1\\Windows
Performance Toolkit\\;C&#8203;:\\Program Files\\Microsoft SQL
Server\\110\\Tools\\Binn\\;C&#8203;:\\Program Files \(x86\)\\Microsoft
SDKs\\TypeScript\\1\.0\\;C&#8203;:\\Strawberry\\c\\bin;C&#8203;:\\Strawberry\\perl\\site\\bin;C&#8203;:\\Strawberry\\perl\\bin;C&#8203;:\\SQLite;C&#8203;:\\Program
Files \(x86\)\\Haskell Platform\\2013\.2\.0\.0\\mingw\\bin;C&#8203;:\\HTML
Tidy;C&#8203;:\\Program Files\\Microsoft Windows Performance
Toolkit\\;C&#8203;:\\Python27;C&#8203;:\\Python27\\Scripts;C&#8203;:\\Program Files
\(x86\)\\Perforce;C&#8203;:\\Program Files\\Haskell
Platform\\2014\.2\.0\.0\\mingw\\bin;C&#8203;:\\Program Files \(x86\)\\Common
Files\\Seagate\\SnapAPI\\;C&#8203;:\\Sysinternals;C&#8203;:\\Program
Files\\WinRAR;C&#8203;:\\Program Files\\Common Files\\Microsoft Shared\\Windows
Live;C&#8203;:\\Program Files \(x86\)\\Common Files\\Hackety
Hack\\0\.r1529\\\.\.;C&#8203;:\\Program Files \(x86\)\\IDM Computer
Solutions\\UltraCompare\\;C&#8203;:\\ffmpeg\\bin;C&#8203;:\\Program Files \(x86\)\\Microsoft
Visual Studio 10\.0\\VC\\bin\\x86\_amd64;C&#8203;:\\Program Files \(x86\)\\Microsoft
Visual Studio 10\.0\\VC\\bin\\amd64;C&#8203;:\\SWIG;C&#8203;:\\MinGW\\bin;C&#8203;:\\MinGW\\MSYS\\1\.0\\bin;C&#8203;:\\Python27;C&#8203;:\\Curl;C&#8203;:\\pkg\-config\\bin;C&#8203;:\\glib\\bin;C&#8203;:\\gettext\-runtime\\bin;C&#8203;:\\Program
Files \(x86\)\\IDM Computer
Solutions\\UltraEdit\\;C&#8203;:\\LuaRocks\\2\.0;C&#8203;:\\Program Files
\(x86\)\\Android\\android\-sdk\\platform\-tools;C&#8203;:\\Ruby193\\bin;C&#8203;:\\Program
Files \(x86\)\\Nmap;C&#8203;:\\MobiPerl;E&#8203;:\\Perl\\source;C&#8203;:\\Program
Files\\R\\R\-3\.1\.2\\bin;C&#8203;:\\Ruby\.DevKit\\bin;C&#8203;:\\Program Files
\(x86\)\\EaseUS\\Todo Backup\\bin\\x64\\
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    PERL\_SOURCE=E&#8203;:\\Perl\\source
&nbsp;    SHELL \(unset\)

**Comment by p5pRT** [comment]
### From @karenetheridge

On Mon Apr 27 09&#8203;:04&#8203;:35 2015\, the\.rob\.dixon@&#8203;gmail\.com wrote&#8203;:
> Clearly the accuracy of anything but the most recent version of
> perlvar isn't a valid topic for a perlbug\, but it would be nice if the
> next version clarified the true situation\.

As also discussed in RT\#124404\, $\] hasn't been deprecated\.

I've attached some patches that&#8203;:
1\. move the documentation for $\] out of the deprecated area of the manual\, and
2\. re\-document the $OLD\_PERL\_VERSION alias\.

**Comment by p5pRT** [comment]
### From @karenetheridge

<details><summary>0001-move-the-documentation-for-out-of-the-deprecated-sec.patch</summary>

```diff
From 3d673bf21012375789b360635b91bf236e812165 Mon Sep 17 00:00:00 2001
From: Karen Etheridge <ether@cpan.org>
Date: Mon, 27 Apr 2015 11:38:52 -0700
Subject: [PATCH 1/2] move the documentation for $] out of the deprecated
 section

---
 pod/perlvar.pod | 64 ++++++++++++++++++++++++++++-----------------------------
 1 file changed, 32 insertions(+), 32 deletions(-)

diff --git a/pod/perlvar.pod b/pod/perlvar.pod
index e14d844..bf3c2fa 100644
--- a/pod/perlvar.pod
+++ b/pod/perlvar.pod
@@ -433,6 +433,38 @@ Previously, only child processes received stringified values:
 This happens because you can't really share arbitrary data structures with
 foreign processes.
 
+=item $]
+X<$]>
+
+The revision, version, and subversion of the Perl interpreter, represented
+as a decimal of the form 5.XXXYYY, where XXX is the version / 1e3 and YYY
+is the subversion / 1e6.  For example, Perl v5.10.1 would be "5.010001".
+
+This variable can be used to determine whether the Perl interpreter
+executing a script is in the right range of versions:
+
+    warn "No PerlIO!\n" if $] lt '5.008';
+
+When comparing C<$]>, string comparison operators are B<highly
+recommended>.  The inherent limitations of binary floating point
+representation can sometimes lead to incorrect comparisons for some
+numbers on some architectures.
+
+See also the documentation of C<use VERSION> and C<require VERSION>
+for a convenient way to fail if the running Perl interpreter is too old.
+
+See L</$^V> for a representation of the Perl version as a L<version>
+object, which allows more flexible string comparisons.
+
+The main advantage of C<$]> over C<$^V> is that it works the same on any
+version of Perl.  The disadvantages are that it can't easily be compared
+to versions in other formats (e.g. literal v-strings, "v1.2.3" or
+version objects) and numeric comparisons can occasionally fail; it's good
+for string literal version checks and bad for comparing to a variable
+that hasn't been sanity-checked.
+
+Mnemonic: Is this version of perl in the right bracket?
+
 =item $SYSTEM_FD_MAX
 
 =item $^F
@@ -2275,38 +2307,6 @@ Mnemonic: [ begins subscripts.
 
 Deprecated in Perl v5.12.0.
 
-=item $]
-X<$]>
-
-The revision, version, and subversion of the Perl interpreter, represented
-as a decimal of the form 5.XXXYYY, where XXX is the version / 1e3 and YYY
-is the subversion / 1e6.  For example, Perl v5.10.1 would be "5.010001".
-
-This variable can be used to determine whether the Perl interpreter
-executing a script is in the right range of versions:
-
-    warn "No PerlIO!\n" if $] lt '5.008';
-
-When comparing C<$]>, string comparison operators are B<highly
-recommended>.  The inherent limitations of binary floating point
-representation can sometimes lead to incorrect comparisons for some
-numbers on some architectures.
-
-See also the documentation of C<use VERSION> and C<require VERSION>
-for a convenient way to fail if the running Perl interpreter is too old.
-
-See L</$^V> for a representation of the Perl version as a L<version>
-object, which allows more flexible string comparisons.
-
-The main advantage of C<$]> over C<$^V> is that it works the same on any
-version of Perl.  The disadvantages are that it can't easily be compared
-to versions in other formats (e.g. literal v-strings, "v1.2.3" or
-version objects) and numeric comparisons can occasionally fail; it's good
-for string literal version checks and bad for comparing to a variable
-that hasn't been sanity-checked.
-
-Mnemonic: Is this version of perl in the right bracket?
-
 =back
 
 =cut
-- 
2.3.2


```
</details>

**Comment by p5pRT** [comment]
### From @karenetheridge

<details><summary>0002-add-back-documentation-that-OLD_PERL_VERSION-is-an-a.patch</summary>

```diff
From 0569c1722919252d941aee5d7e8ebba70c237bcc Mon Sep 17 00:00:00 2001
From: Karen Etheridge <ether@cpan.org>
Date: Mon, 27 Apr 2015 11:48:13 -0700
Subject: [PATCH 2/2] add back documentation that $OLD_PERL_VERSION is an alias
 for $]

---
 pod/perlvar.pod | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pod/perlvar.pod b/pod/perlvar.pod
index bf3c2fa..a8c77ec 100644
--- a/pod/perlvar.pod
+++ b/pod/perlvar.pod
@@ -433,6 +433,8 @@ Previously, only child processes received stringified values:
 This happens because you can't really share arbitrary data structures with
 foreign processes.
 
+=item $OLD_PERL_VERSION
+
 =item $]
 X<$]>
 
-- 
2.3.2


```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @khwilliamson

The documentation has now been patched with
4ad0ecd4c5a81ed8f71804b908656dca23f4867e
d2578154934acc58c35260d46623962e12848748
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'open' to 'pending release'

**Comment by p5pRT** [comment]
### From @khwilliamson

Thank you for submitting this ticket\.

The issue should now be resolved with the release today of Perl v5\.22\, which is available at http&#8203;://www\.perl\.org/get\.html
\-\- 
Karl Williamson for the Perl 5 team

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'pending release' to 'resolved'
